File: test_api.js
const fetch = require('node-fetch');

async function test() {
  console.log("Testing API...");
  try {
    const res = await fetch('http://localhost:3000/api/extract', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ engine: 'x', target: 'elonmusk' })
    });
    const text = await res.text();
    console.log("STATUS:", res.status);
    console.log("RESPONSE:", text.substring(0, 500) + '...');
  } catch(e) {
    console.error(e);
  }
}

test();

File: test_api.py
"""Quick test to hit the Flask API and print the raw response."""
import urllib.request
import json
import time

url = "http://localhost:3000/api/extract"
payload = json.dumps({"engine": "tiktok", "target": "tonsic"}).encode("utf-8")
req = urllib.request.Request(url, data=payload, headers={"Content-Type": "application/json"})

print(f"[*] Sending POST to {url}...")
start = time.time()
try:
    resp = urllib.request.urlopen(req, timeout=120)
    body = resp.read().decode("utf-8")
    elapsed = time.time() - start
    print(f"[+] Status: {resp.status}  |  Elapsed: {elapsed:.1f}s  |  Body length: {len(body)} chars")
    print(f"[+] First 800 chars of response:\n{body[:800]}")
except Exception as e:
    elapsed = time.time() - start
    print(f"[!] Error after {elapsed:.1f}s: {e}")
